user_sync/connector/directory_ldap.py

```diff
@@ -269,35 +272,55 @@ def iter_users(self, users_filter):
                 if (last_attribute_name != None):
                     self.logger.warn('No email attribute: %s for dn: %s', last_attribute_name, dn)
                 continue
+
+            source_attributes = {}
             
             user = user_sync.connector.helper.create_blank_user()
             user['email'] = email
+            source_attributes['email'] = email
                 
             username, last_attribute_name = self.user_username_formatter.generate_value(record)
             if (username == None and last_attribute_name != None):
                 self.logger.info('No username attribute: %s for dn: %s', last_attribute_name, dn)    
             user['username'] = username if username != None else email
+            if (username != None):
```

Comment:
May be moot because I think it is always set in user results now, but seems like leaving it out when there is no value is reasonable vs having it present with a None value.